### PR TITLE
web: Adds support for base embed/object attribute (Part of #4258)

### PR DESCRIPTION
**Pull request by hatal175**

### CODE REVIEW

**Comment by hatal175** [comment]
I would like to comment that current;y the base value requires to have a slash at the end of the url because of url::parse. I'm not sure what the original flash behavior was.

File: web/packages/core/src/load-options.ts [hunk 1]
```diff
@@ -178,6 +178,14 @@ export interface BaseLoadOptions {
         secs: number;
         nanos: number;
     };
+
+    /**
+     * Specifies the base directory or URL used to resolve all relative path statements in the SWF file.
+     * null means the current dirsctory.
```
Comment (kmeisthax) [inline web/packages/core/src/load-options.ts:184]: spelling: "directory", not "dirsctory"

**Review by kmeisthax — CHANGES_REQUESTED** [review]
I have one minor nitpick involving spelling on a doc comment, but everything else looks fine.

File: web/src/navigator.rs [hunk 2]
```diff
@@ -33,6 +40,17 @@ impl WebNavigatorBackend {
             performance,
             allow_script_access,
             upgrade_to_https,
+            base_url,
+        }
+    }
+
+    fn base_uri(&self, document: &Document) -> Option<String> {
```
Comment (adrian17) [inline web/src/navigator.rs:47]: Ideally this should probably return a `Cow`, to avoid copying the base_url... but that doesn't really matter much here.

**Review by adrian17 — APPROVED** [review]
Before merging: did you test this on any specific existing content using this attribute, or did you make your own sample page?

**Comment by hatal175** [comment]
Hard to find content so I made my own sample page.

**Comment by adrian17** [comment]
OK.

**Comment by adrian17** [comment]
Merged. Thanks!

**Comment by albertstill** [comment]
Thanks for working on this.

I couldn't actually get it to work in my flash games, I debugged and saw the `base` field being passed into the Ruffle instance here correctly, via `config`, as expected.

https://github.com/ruffle-rs/ruffle/blob/3982f3af9811cfc32c6f7eaf06aa3c8c81b91226/web/packages/core/src/ruffle-player.ts#L417

But it never altered the subsequent swf files my initial swf requested, I could never get the urls to reflect this base property.

Am I expecting the correct behaviour?

I'm using self hosted nightly 2021-09-04 with the polyfill approach.

Thanks

**Comment by hatal175** [comment]
Mind sending the html snippet?

**Comment by albertstill** [comment]
Thanks @hatal175, here is a [snippet](https://github.com/albertstill/ruffle-base-issue/blob/master/index.html#L13-L29) and repo where you can reproduce this issue, thanks.

https://github.com/albertstill/ruffle-base-issue

**Pull request merged**